app/screens/interactive_dialog/interactive_dialog.test.js

```diff
@@ -106,4 +106,38 @@ describe('InteractiveDialog', () => {
         expect(baseProps.actions.submitInteractiveDialog).toHaveBeenCalledTimes(1);
         expect(baseProps.actions.submitInteractiveDialog).toHaveBeenCalledWith(dialog);
     });
+
+    describe('generic error message', () => {
+        test('should show error when submit returns an error', async () => {
+            const props = {
+                ...baseProps,
+                actions: {
+                    ...baseProps.actions,
+                    submitInteractiveDialog: async () => ({
+                        data: {error: 'This is an error message.'},
+                    }),
+                },
+            };
+
+            const wrapper = shallow(
+                <InteractiveDialog {...props}/>
+            );
+            wrapper.instance().scrollView = {current: {scrollTo: jest.fn()}};
+
+            await wrapper.instance().handleSubmit();
+            expect(wrapper.state().error).toEqual('This is an error message.');
```

Comment:
I tried to find examples of tests that check the actual rendered components, and I could only find snapshot examples.  When I tried to log the rendered components, it would not show the error message at all.  I am not experienced in React Native, but my research suggests that this is a current limitation of React Native testing.

I went with checking the state as a starting point.  Is there a better approach I could take?

Reply:
you could do something like `expect(wrapper.find(ErrorText).length).toBe(0)`

Reply:
Done.  It would be nice to be able to assert on the actual error message being rendered, since that is what the user will see, but that does not seem to render with `shallow`, and I was not able to get `mount` to work (see comment above).

Maybe having the `state.error` and the `ErrorText` checks are a reasonable compromise?

Reply:
@migbot , that worked.  Thank you.